src/Bicep.LangServer/Handlers/BicepDefinitionHandler.cs

```diff
@@ -2,58 +2,244 @@
 // Licensed under the MIT License.
 using System.Threading;
 using System.Threading.Tasks;
+using Bicep.Core.FileSystem;
 using Bicep.Core.Semantics;
+using Bicep.Core.Syntax;
+using Bicep.LanguageServer.CompilationManager;
+using Bicep.LanguageServer.Completions;
 using Bicep.LanguageServer.Extensions;
 using Bicep.LanguageServer.Providers;
 using Bicep.LanguageServer.Utils;
 using OmniSharp.Extensions.LanguageServer.Protocol.Client.Capabilities;
 using OmniSharp.Extensions.LanguageServer.Protocol.Document;
 using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using System;
+using Bicep.Core.Parsing;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Bicep.Core;
+using Range = OmniSharp.Extensions.LanguageServer.Protocol.Models.Range;
+using System.Linq;
 
 namespace Bicep.LanguageServer.Handlers
 {
     public class BicepDefinitionHandler : DefinitionHandlerBase
     {
         private readonly ISymbolResolver symbolResolver;
 
-        public BicepDefinitionHandler(ISymbolResolver symbolResolver) : base()
+        private readonly ICompilationManager compilationManager;
+
+        private readonly IFileResolver fileResolver;
+
+        public BicepDefinitionHandler(ISymbolResolver symbolResolver, ICompilationManager compilationManager, IFileResolver fileResolver) : base()
         {
             this.symbolResolver = symbolResolver;
+            this.compilationManager = compilationManager;
+            this.fileResolver = fileResolver;
         }
 
         public override Task<LocationOrLocationLinks> Handle(DefinitionParams request, CancellationToken cancellationToken)
```

Comment:
I have noticed that every `ctrl + click` calls this _twice_ (from what i see its identical request). Any ideas why that could be the case?

Reply:
Is the behavior different for F12?

This is a random guess, but it's possible that VS code is sending the first request on Ctrl+hover to determine if the link can even go anywhere and then sending the 2nd one on ctrl+click.

Reply:
You nailed it. The first request is responsible for `ctrl+hover`, that's why when you hover it the valid content is underlined.